mangaki/mangaki/models.py

```diff
@@ -450,6 +450,28 @@ def __str__(self):
             self.pk, self.user, self.work.title, self.get_problem_display()
         )
 
+    @property
+    def can_auto_fix(self):
+        # FIXME: use Enum + dynamic based on evidences / message (links).
+        return self.problem in ('nsfw', 'n_nsfw')
+
+    @transaction.atomic
+    def auto_fix(self):
+        """
+        Apply automatically a fix on the issue.
+        e.g. for a NSFW (resp. non-NSFW) problem, it'll set the work as NSFW (resp. non-NSFW).
+
+        It'll raise ValueError when it is impossible to automatically fix the issue.
+        """
+        if self.problem in ('nsfw', 'n_nsfw'):
+            self.work.nsfw = not self.work.nsfw
```

Comment:
What happened is, I checked a NSFW but it was also put to NSFW (maybe manually by someone else). So it inverted it. Please make two if statements, regarding the value of `self.problem`!

Reply:
Strange, but will do.

Reply:
Je vais réexpliquer en français.
Ton bouton inverse l'état NSFW/SFW.
Mais du coup, la suggestion disait : « Ceci doit être NSFW » mais ça l'était déjà (genre, quelqu'un l'avait fait manuellement depuis l'admin). Du coup, en cliquant sur la suggestion, ça a retiré le bandeau NSFW. Donc le code n'a pas le bon comportement, il devrait obéir à la suggestion, pas défaire un travail fait !

Reply:
C'est fait :).